lib/Models/addUserFiles.js

```diff
@@ -23,28 +25,42 @@ add data from a file on your system.  Please upgrade your web browser.  For the
     }
 
     const promises = [];
+    const tempCatalogItemList = [];
 
     for (let i = 0; i < files.length; ++i) {
         const file = files[i];
+        const tempCatalogItem = new AsyncFunctionResultCatalogItem(terria);
+        tempCatalogItem.name = file.name;
+        tempCatalogItem.description = "Loading file...";
+        terria.catalog.userAddedDataGroup.add(tempCatalogItem);
+        terria.catalog.userAddedDataGroup.isOpen = true;
 
         terria.analytics.logEvent('uploadFile', 'browse', file.name);
 
+        let loadPromise;
         if (file.name.toUpperCase().indexOf('.JSON') !== -1) {
             const promise = readJson(file).then(json => {
                 if (json.catalog || json.services) {
                     // This is an init file.
                     return terria.addInitSource(json);
                 }
-                return addUserCatalogMember(terria, createCatalogItemFromFileOrUrl(terria, viewState, file, dataType.value, true));
+                loadPromise = addUserCatalogMember(terria, createCatalogItemFromFileOrUrl(terria, viewState, file, dataType.value, true));
             });
-            promises.push(raiseErrorOnRejectedPromise(terria, promise));
+            loadPromise = raiseErrorOnRejectedPromise(terria, promise);
+            promises.push(loadPromise);
         } else {
-            promises.push(addUserCatalogMember(terria, createCatalogItemFromFileOrUrl(terria, viewState, file, dataType.value, true)));
+            loadPromise = addUserCatalogMember(terria, createCatalogItemFromFileOrUrl(terria, viewState, file, dataType.value, true));
+            promises.push(loadPromise);
         }
+        tempCatalogItem.loadPromise = loadPromise;
+        tempCatalogItem.isEnabled = true;
+        tempCatalogItemList.push(tempCatalogItem);
     }
 
     return when.all(promises, addedItems => {
-        return addedItems.filter(item => item && !(item instanceof TerriaError));
+        const items = addedItems.filter(item => item && !(item instanceof TerriaError));
+        tempCatalogItemList.map(function(value) { terria.catalog.userAddedDataGroup.remove(value); });
```

Comment:
`forEach` would be a little cleaner than `map` here, since you're not using the result at all.